pymc3/step_methods/smc.py

```diff
@@ -147,6 +147,10 @@ def __init__(self, vars=None, out_vars=None, n_chains=100, scaling=1., covarianc
         vars = inputvars(vars)
 
         if out_vars is None:
+            if not any(likelihood_name == RV.name for RV in model.unobserved_RVs):
```

Comment:
Each step method can set `generates_stats` to True and define dtypes for the individual values in `stats_dtypes`. You can have a look at `nuts.py` lines 75 to 87. The step method must then return for each step a tuple `(q, [stats])`, where `q` is the sampled point and `stats` contains the statistics it wants to store.